offline/rabbitmq_setup.md

```diff
@@ -0,0 +1,149 @@
+```
+d helm upgrade --install rabbitmq ./charts/rabbitmq --values ./values/rabbitmq/values.yaml --values ./values/rabbitmq/secrets.yaml
+```
+
+### Method 2: Install RabbitMQ outside of the Kubernetes cluster with an Ansible playbook
+
+Add the nodes on which you want to run rabbitmq to the `[rmq-cluster]` group in the `ansible/inventory/offline/hosts.ini` file. Also, update the `ansible/roles/rabbitmq-cluster/defaults/main.yml` file with the correct configurations for your environment.
+
+If you need RabbitMQ to listen on a different interface than the default gateway, set `rabbitmq_network_interface`
+
+You should have following entries in the `/ansible/inventory/offline/hosts.ini` file. For example:
+```
+[rmq-cluster:vars]
+rabbitmq_network_interface = enp1s0
+
+[rmq-cluster]
+ansnode1
+ansnode2
+ansnode3
+```
+
+#### Hostname Resolution
+RabbitMQ nodes address each other using a node name, a combination of a prefix and domain name, either short or fully-qualified (FQDNs). For e.g. rabbitmq@ansnode1
+
+Therefore every cluster member must be able to resolve hostnames of every other cluster member, its own hostname, as well as machines on which command line tools such as rabbitmqctl might be used.
+
+Nodes will perform hostname resolution early on node boot. In container-based environments it is important that hostname resolution is ready before the container is started.
+
+Hostname resolution can use any of the standard OS-provided methods:
+
+For e.g. DNS records
+Local host files (e.g. /etc/hosts)
+Reference - https://www.rabbitmq.com/clustering.html#cluster-formation-requirements
+
+
+For adding entries to local host file(`/etc/hosts`), run
+```
+d ansible-playbook -i ansible/inventory/offline/hosts.ini ansible/roles/rabbitmq-cluster/tasks/configure_dns.yml
+```
+
+Create the rabbitmq cluster:
+
+``` 
+d ansible-playbook -i ansible/inventory/offline/hosts.ini ansible/rabbitmq.yml
+```
+
+and run the following playbook to create values file for helm charts to look for RabbitMQ IP addresses -
+
+```
+d ansible-playbook -i ./ansible/inventory/offline/hosts.ini ansible/helm_external.yml --tags=rabbitmq-external
+```
+
+Make Kubernetes aware of where RabbitMQ external stateful service is running:
+```
+d helm install rabbitmq-external ./charts/rabbitmq-external --values ./values/rabbitmq-external/values.yaml
+```
+
+Configure wire-server to use the external RabbitMQ service:
+
+Edit the `/values/wire-server/prod-values.example.yaml` file to update the RabbitMQ host
+Under `brig` and `galley` section, you will find the `rabbitmq` config, update the host to `rabbitmq-external`, it should look like this:
+```
+rabbitmq:
+  host: rabbitmq-external
+``` 
+
+## Backup and Restore
+
+Following steps describe the backup and restore process for RabbitMQ deployed outside of Kubernetes.
+
+Although, this can vary based on your setup, it is also recommended to follow the official documentation here - https://www.rabbitmq.com/docs/backup
```

Comment:
```suggestion
This can vary based on your security, privacy, and administrative policies. It is also recommended to read and follow the official documentation here - https://www.rabbitmq.com/docs/backup
```